src/core/Akka.Streams.Tests/Dsl/FlowLogSpec.cs

```diff
@@ -26,145 +28,169 @@ public FlowLogSpec(ITestOutputHelper helper) : base("akka.loglevel = DEBUG", hel
                 .Log("flow-s3", t => t.Item2)
                 .RunWith(Sink.Ignore<(int, string)>(), Materializer);
 
-            var msgs = LogMessages(2);
+            var msgs = await LogMessages(2, "[flow-s3]");
             msgs[0].Should().Be("[flow-s3] Element: 42");
             msgs[1].Should().Be("[flow-s3] Upstream finished.");
         }
 
         [Fact]
-        public void A_Log_on_source_must_log_upstream_failure()
+        public async Task A_Log_on_source_must_log_upstream_failure()
         {
             var cause = new TestException("test");
-            Source.Failed<int>(cause)
+            _ = Source.Failed<int>(cause)
                 .Log("flow-s4")
                 .RunWith(Sink.Ignore<int>(), Materializer);
 
-            var error = LogProbe.ExpectMsg<Error>();
+            var error = (Error) await _logProbe.FishForMessageAsync(o => o is Error e && e.Message.ToString().StartsWith("[flow-s4]"));
             error.Cause.Should().Be(cause);
             error.Message.ToString().Should().Be("[flow-s4] Upstream failed.");
         }
 
         [Fact]
-        public void A_Log_on_source_must_allow_passing_in_custom_LoggingAdapter()
+        public async Task A_Log_on_source_must_allow_passing_in_custom_LoggingAdapter()
         {
             var log = new BusLogging(Sys.EventStream, "com.example.ImportantLogger", LogType, DefaultLogMessageFormatter.Instance);
 
-            Source.Single(42)
+            _ = Source.Single(42)
                 .Log("flow-5", log: log)
                 .RunWith(Sink.Ignore<int>(), Materializer);
 
-            var msgs = LogProbe.ReceiveN(2).Cast<Debug>().ToArray();
+            var msgs = await LogEvents<Debug>(2, "[flow-5]");
             msgs.All(m => m.LogSource.Equals("com.example.ImportantLogger")).Should().BeTrue();
             msgs.All(m => m.LogClass == LogType).Should().BeTrue();
             msgs[0].Message.ToString().Should().Be("[flow-5] Element: 42");
             msgs[1].Message.ToString().Should().Be("[flow-5] Upstream finished.");
         }
 
         [Fact]
-        public void A_Log_on_source_must_allow_configuring_log_levels_via_Attributes()
+        public async Task A_Log_on_source_must_allow_configuring_log_levels_via_Attributes()
         {
             var logAttributes = Attributes.CreateLogLevels(LogLevel.WarningLevel, LogLevel.InfoLevel,
                 LogLevel.DebugLevel);
 
-            Source.Single(42)
+            _ = Source.Single(42)
                 .Log("flow-6")
                 .WithAttributes(Attributes.CreateLogLevels(LogLevel.WarningLevel, LogLevel.InfoLevel,
                     LogLevel.DebugLevel))
                 .RunWith(Sink.Ignore<int>(), Materializer);
 
-            LogProbe.ExpectMsg<Warning>().Message.ToString().Should().Be("[flow-6] Element: 42");
-            LogProbe.ExpectMsg<Info>().Message.ToString().Should().Be("[flow-6] Upstream finished.");
+            var warnings = await LogEvents<Warning>(1, "[flow-6]");
```

Comment:
`ExpectMsg` replaced with the helper function `LogEvents`. We want to make sure that we're not reading any event bus noise